### PR TITLE
Certify Oracle Enterprise Linux 7

### PR DESCRIPTION
**Pull request by frankgh**
- Add a docker image to build an OEL7 image
- Add a job to certify PXF against OEL7 running Greenplum 6

### CODE REVIEW

**Comment by frankgh** [comment]
Green pipelines: https://ud.ci.gpdb.pivotal.io/teams/main/pipelines/dev:fguerrero-certify-oel7

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 1]
```diff
@@ -556,6 +565,34 @@ jobs:
...
+    file: pxf_src/concourse/tasks/test.yml
+    image: gpdb6-pxf-dev-oel7-image
+    params:
+      ACCESS_KEY_ID: ((tf-machine-access-key-id))
+      GP_VER: 6
+      GROUP: gpdb,proxy,profile
+      SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
+      TARGET_OS: centos
+      TARGET_OS_VERSION: 7
+
```
Comment (oliverralbertini) [inline concourse/pipelines/templates/build_pipeline-tpl.yml:595]: Don't forget to add this to the list of passed items in the testing gate.

File: concourse/docker/pxf-dev-base/gpdb6/oel7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    # create .pxfrc
+    && echo >> ~gpadmin/.pxfrc 'export LANG=en_US.UTF-8' \
+    && echo >> ~gpadmin/.pxfrc 'export PGPORT=5432' \
+    && echo >> ~gpadmin/.pxfrc 'export GOPATH=/opt/go' \
+    && echo >> ~gpadmin/.pxfrc 'export GPHOME=$(find /usr/local/ -name greenplum-db* -type d | head -n1)' \
+    && echo >> ~gpadmin/.pxfrc 'export LD_LIBRARY_PATH=${GPHOME}/lib:${LD_LIBRARY_PATH-}' \
+    && echo >> ~gpadmin/.pxfrc 'export GPHD_ROOT=/singlecluster' \
+    && echo >> ~gpadmin/.pxfrc 'export PXF_HOME=/usr/local/pxf-gp6' \
+    && echo >> ~gpadmin/.pxfrc 'export PXF_CONF=/home/gpadmin/pxf' \
+    && echo >> ~gpadmin/.pxfrc 'export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk' \
```
Comment (oliverralbertini) [inline concourse/docker/pxf-dev-base/gpdb6/oel7/Dockerfile:60]: I guess the JVM wasn't an issue?

File: concourse/pipelines/templates/build_pipeline-tpl.yml (hunk 1 above)
Comment (frankgh) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:599]: done!

File: concourse/docker/pxf-dev-base/gpdb6/oel7/Dockerfile (hunk 2 above)
Comment (frankgh) [reply concourse/docker/pxf-dev-base/gpdb6/oel7/Dockerfile:60]: a Java 8 is installed in the base image, so it was not an issue

**Comment by frankgh** [comment]
@oliverralbertini yeah, we can decide on the PR pipeline. let me know what you prefer

**Comment by oliverralbertini** [comment]
> @oliverralbertini yeah, we can decide on the PR pipeline. let me know what you prefer

The weird part about it is that the PR pipeline builds/tests in the same job. So it's not clear where the OEL test would go.

**Comment by oliverralbertini** [comment]
Do we need to test backwards compatibility for OEL7?

**Comment by frankgh** [comment]
> Do we need to test backwards compatibility for OEL7?

Probably not. It was sufficient to do a one-off test or CI. I think we are doing enough to test. I'll leave it up to you.

File: concourse/docker/pxf-dev-base/README.md [hunk 3]
```diff
@@ -55,3 +55,16 @@ command to build the image:
...
     popd
+
+### Docker gpdb6-oel7-test-pxf-image image
+
+Build this image for Greenplum 6 running Oracle Enterprise Linux 7. Run the
+following command to build the image:
+
+    pushd ~/workspace/pxf/concourse/docker/pxf-dev-base/
+    docker build \
+      --build-arg=BASE_IMAGE=pivotaldata/gpdb-dev:latest \
```
[the remark this answers is not in the document]
Comment (frankgh) [reply concourse/docker/pxf-dev-base/README.md:66]: I need to fix it :)

**Pull request merged**